internal-functions

```diff
@@ -128,6 +128,14 @@ fn-query-status(){
     AUTH_URL="$AUTH_URL/Auth"
   fi
 
-  local USER_INFO_URL="https://$AUTH_URL/v2/users/$USER/"
-  curl -sk --max-time 3 --header "$HEADER" "$USER_INFO_URL" | jq ".$FIELD" | grep -q 'true'
+  local USER_INFO_URL="https://$AUTH_URL/v2/users/$USER/"  
+  RESPONSE=$(curl -sS --insecure --max-time 10 --retry 5 --header "$HEADER" --write-out "\n%{http_code}" "$USER_INFO_URL")
+  RESPONSE_BODY=$(echo "$RESPONSE" | awk "NR==1")
+  STATUS_CODE=$(echo "$RESPONSE" | awk "NR==2")
+
+  if [[ "$STATUS_CODE" != "200" ]]; then
```

Comment:
I wonder if it would be better to be a little more generic with our error message here given that this request may fail for reasons other than a timeout. @tarzzz -- thoughts?

Reply:
The error message is actually recommended by Chris here: https://github.com/plotly/streambed/issues/14120#issuecomment-587086730

Reply:
We can discuss it in the issue thread if you feel like we can improve the message itself :)

Reply:
Right - it seems to me that the suggestion was more general with the inclusions of the "might have": "your system might be overloaded and the authentication check _might have_ timed out." Maybe, instead of changing the error message as you have it, we can explicitly check for a timeout status as opposed to a generic "!=200" - thoughts ?

Reply:
Hm, We don't exactly know which "code" causes the intermittent error described in: https://github.com/plotly/streambed/issues/14120

This PR will "probably" fix that error, but we can't be sure until we deploy it on playground for some time.

One way so that the error message is more accurate to the situation is to only check for `503` and `504`:

> **503 Service Unavailable**
The server cannot handle the request (because it is overloaded or down for maintenance). Generally, this is a temporary state.
**504 Gateway Timeout**
The server was acting as a gateway or proxy and did not receive a timely response from the upstream server.

Thoughts?

Reply:
Edit: We should include "502 -- Connection refused" to cover most of the possibilities..

Reply:
Handling the 50X errors separately seems like a nice approach! The 503 error message you suggest looks great to me. Perhaps the 504 error message can be simplified if it's intended for clients with varying levels of tech experience. Maybe @chriddyp has some thoughts on this as well. 

Two things also come to mind: 

1)  Perhaps we also want a catch for all other non-200 error codes as well with a generic message so that it doesn't simply bubble up with a misleading error message (as was seen in the issue)

2) If handling 50X error messages separately is overkill for now, I think just going with a general error message (not locking it down to a timeout) could work as well.

Reply:
>  other non-200 error codes as well with a generic message so that it doesn't simply bubble up with a misleading error message (as was seen in the issue)

Yeah, like: `"A {} level error occurred during an authentication check. Try again in a little bit or reach out to onpremise.support@plot.ly".format(STATUS_CODE)`

Then, at least we'll know what error code was returned without going through the support bundle process.

Then, for the timeout issues, we can include the explicit "your system might be overloaded" message.